renderlib/io/FileReaderZarr.cpp

```diff
@@ -123,7 +123,35 @@ FileReaderZarr::jsonRead(const std::string& zarrurl)
       attrs = ::nlohmann::json::object_t();
     }
   }
+  return attrs;
+}
+
+::nlohmann::json
+FileReaderZarr::jsonRead(const std::string& zarrurl)
+{
+  if (m_zattrs.is_object()) {
+    return m_zattrs;
+  }
+
+  bool hasV3 = false;
+
+  // try zarr.json first (indicating Zarr v3)
+  // and then try .zattrs (indicating Zarr v2)
+  nlohmann::json attrs = tryReadJson(zarrurl, "zarr.json");
+  if (attrs.is_object() && !attrs.empty()) {
+    hasV3 = true;
+  } else {
+    attrs = tryReadJson(zarrurl, ".zattrs");
+  }
+  if (attrs.is_object() && !attrs.empty()) {
+    LOG_DEBUG << (hasV3 ? "Zarr v3" : "Zarr v2");
```

Comment:
This log seem a bit unintuitive to me as a non-frequent user, especially this check is identical to the one on L141. Does it indicate that it's loading V3/V2 Zarr metadata from the given `zarrurl`?

Reply:
the value of attrs has changed here because we tried to load json from a different filename (zarr.json vs .zattrs). that's why the check happens a second time whether or not the first time failed.
After we have tried twice, we know all we need to know to continue.
Based on Peyton's comment, I think I will clean this up a little bit.